spec/lib/reports/customers_report_spec.rb

```diff
@@ -140,30 +140,62 @@ module Customers
                 end
 
-                it "returns one row per customer per shipping method" do
-                  expect(subject.query_result.size).to eq(2)
-                  expect(subject.table_rows.size).to eq(2)
-                  expect(subject.table_rows).to eq([[
-                                                     a.firstname, a.lastname,
-                                                     [a.address1, a.address2, a.city].join(" "),
-                                                     o1.email, a.phone, d.name,
-                                                     [d.address.address1, d.address.address2, d.address.city].join(" "),
-                                                     o1.shipping_method.name
-                                                   ], [
-                                                     a.firstname, a.lastname,
-                                                     [a.address1, a.address2, a.city].join(" "),
-                                                     o2.email, a.phone, d.name,
-                                                     [d.address.address1, d.address.address2, d.address.city].join(" "),
-                                                     sm2.name
-                                                   ]])
+                context "when the shipping method column is being included" do
+                  let(:fields_to_show) do
+                    [:first_name, :last_name, :billing_address, :email, :phone, :hub, :hub_address,
+                     :shipping_method]
+                  end
+                  subject { Addresses.new(user, { fields_to_show: fields_to_show }) }
+
+                  it "returns one row per customer per shipping method" do
+                    expect(subject.query_result.size).to eq(2)
+                    expect(subject.table_rows.size).to eq(2)
+                    expect(subject.table_rows).to eq([[
+                                                       a.firstname, a.lastname,
+                                                       [a.address1, a.address2, a.city].join(" "),
+                                                       o1.email, a.phone, d.name,
+                                                       [d.address.address1, d.address.address2, d.address.city].join(" "),
+                                                       o1.shipping_method.name
+                                                     ], [
+                                                       a.firstname, a.lastname,
+                                                       [a.address1, a.address2, a.city].join(" "),
+                                                       o2.email, a.phone, d.name,
+                                                       [d.address.address1, d.address.address2, d.address.city].join(" "),
+                                                       sm2.name
+                                                     ]])
```

Comment:
I am nitpicking a little bit, but could you reformat this as you did below ? much more readable !